File: modules/11-testing/swift/Sources/Normalize/Normalize.swift
// Module 11 in Swift: the same Normalize subject. Tests use swift-testing
// (@Test, #expect, and parameterized @Test(arguments:)); the production tools
// for property and fuzz are swift-testing's arguments, SwiftCheck, and libFuzzer
// via SwiftPM -- see the README.

public func normalize(_ s: String) -> String {
  s.split(whereSeparator: { $0.isWhitespace })
    .joined(separator: " ")
    .lowercased()
}

File: modules/11-testing/kotlin/lib/Normalize.kt
// Module 11 in Kotlin: the same Normalize subject. Tests use plain check() to
// stay dependency-free (kotlinc, no Gradle); the production tools are
// kotlin.test / JUnit (units), kotest property testing, and jazzer/JQF
// (fuzzing) -- see the README.

fun normalize(s: String): String =
    s
        .split(Regex("\\s+"))
        .filter { it.isNotEmpty() }
        .joinToString(" ")
        .lowercase()
